File: app/src/main/java/com/binatestation/kickstart/adapters/ImageViewFlipperAdapter.java
package com.binatestation.kickstart.adapters;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;

import com.binatestation.kickstart.R;
import com.binatestation.kickstart.models.BaseModel;
import com.binatestation.kickstart.utils.Utils;

import java.util.ArrayList;


/**
 * Created by RKR on 30-08-2017.
 * ImageViewFlipperAdapter
 */

public class ImageViewFlipperAdapter extends BaseAdapter {

    private ArrayList<Object> data = new ArrayList<>();

    public ImageViewFlipperAdapter() {
    }


    public void setData(ArrayList<?> data) {
        this.data = new ArrayList<>(data);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return data.size();
    }

    @Override
    public Object getItem(int position) {
        return data.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @SuppressLint("ViewHolder")
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        convertView = LayoutInflater.from(parent.getContext()).inflate(R.layout.adapter_image_switcher, parent, false);
        if (position < data.size()) {
            Object object = getItem(position);
            if (object instanceof BaseModel) {
                BaseModel baseModel = (BaseModel) object;

                ImageView image = convertView.findViewById(R.id.image_view);
                Utils.setProfileImage(image, baseModel.getName());
            }
        }

        return convertView;
    }
}

File: kickoff/src/main/java/com/binatestation/android/kickoff/utils/adapters/ImageViewFlipperAdapter.kt
/*
 * (c) Binate Station Private Limited. All rights reserved.
 */

@file:Suppress("unused")

package com.binatestation.android.kickoff.utils.adapters

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.binatestation.android.kickoff.R
import java.util.*


/**
 * Created by RKR on 30-08-2017.
 * ImageViewFlipperAdapter
 */

class ImageViewFlipperAdapter : BaseAdapter() {

    private var data = ArrayList<Any>()

    fun setData(data: ArrayList<*>) {
        this.data = ArrayList(data)
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return data.size
    }

    override fun getItem(position: Int): Any {
        return data[position]
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    @SuppressLint("ViewHolder")
    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        return LayoutInflater.from(parent.context)
            .inflate(R.layout.adapter_image_switcher, parent, false)
    }
}
